org-code-javabuilder/README.md

```diff
@@ -1,33 +1,138 @@
+   http://localhost-studio.code.org:3000/projects/javalab/new
 1. Click the "Run" button
 
 Alternatively, you can test in "headless" mode
 1. Install wscat: `npm install -g wscat`
-1. Launch the WebSocketServer: `gradle appRun`
-1. In a terminal, connect to the server with wscat: `wscat -c ws://localhost:8080/javabuilder`
+1. Launch the WebSocketServer: `./gradlew appRun`
+1. In a terminal, connect to the server with wscat: 
+   `wscat -c ws://localhost:8080/javabuilder`
+
+Postman is a good wscat alternative. It is in beta, but has a gui and is more
+feature-rich.
+
+### Developing with AWS
+For development purposes, you generally shouldn't need to deploy Javabuilder. Some cases
+that do need a dev deployment of Javabuilder:
+* Editing the API Gateway Route Lambdas.
+* Editing the Javabuilder Authorizer Lambda.
+* Editing the Javabuilder Cloud Formation template or deploy script.
+* Editing the Lambda-specific portions of the build and run Lambda. These are all
+  prefixed with `AWS` or `Lambda` in the org-code-javabuilder directory.
+
+**Deploying Javabuilder**  
+To deploy a development version of Javabuilder, see the
+[Deploying Javabuilder](https://docs.google.com/document/d/1mMQK6HhniLsz9lynzhUcm7Tcw_2WVLBxADe0WzqL6rM/edit#)
+instruction doc.
+
+Once you have deployed Javabuilder, you can run a basic connection to it with
+```
+wscat -c wss://<host-name>/?Authorization=connectivityTest
+```
+**Developing with deployed Javabuilder**  
+If you need to test against a level's Java code, use the following steps:
+1. Log in to https://staging-studio.code.org 
+1. Navigate to the level or project you'd like to test against
+1. Open the network tab on your browser developer tools
+1. Click "Run" in Javalab
+1. Look for the network request to 
+   https://staging-studio.code.org/javabuilder/access_token?projectUrl=...
+   in the response, there will be a token. Copy the entire token, without quotes around
+   it.
+1. Open up Postman. Choose new-> WebSocket Request 
+1. Put the url of your javabuilder instance in the server url section (such as 
+   wss://javabuilder-myname.dev-code.org)
+1. Under Params add the key `Authorization` with the value of the token you copied earlier
+1. Under Headers add the key `Origin` with the value https://staging-studio.code.org
+1. Click connect
+
+You should now start seeing messages from Javabuilder! Your token will last for 15
+minutes. If you make changes to the code, you do not need to re-generate your token.
 
-### Developing with AWS (under construction)
-In order to test against the AWS services, you'll need to integrate your IDE with AWS SAM.
+**Developing with deployed Javabuilder and an adhoc environment**  
+1. Deploy javabuilder with the instructions above
+1. In the code-dot-org repo, edit the `javabuilder_url` value in 
+   [cdo.rb](https://github.com/code-dot-org/code-dot-org/blob/3219e5866689117e086d9891effe0fb39b9ae3f0/lib/cdo.rb#L131)
+   to point to your local dev deployment.
```

Comment:
you also need to add it to `upgrade_insecure_requests` [example](https://github.com/code-dot-org/code-dot-org/commit/945fa3ad38be6d85cb7c7aaeda5b3bf2e0fde60c#diff-19cc5be92c36ff06b63767f0ff922d2b9b7b9b8bebe4eaf38e0f331a14b0b528R53)

Reply:
👍